Spreadsheet/main.js

```diff
@@ -1,109 +1,48 @@
-        let x = cols[i].children;
-        for(let j = 0; j < x.length; j++) {
-            if(x[j].classList.value !== 'box parentBoxColor' || x[j].classList.value !== 'box box-border') {
-                return
-            } else {
-                x[j].classList.remove('parentBoxColor');
-                x[j].classList.remove('box-border');
-            }
-        }
-    }; 
-};
-*/
-
-function select(e) {
-    const selectedBoxId = e.path[0].id;
-    const x = document.getElementById('numbers').children;
-    //console.log(e.path[1]);        
-    if(e.path[0].id === '0') {
-        e.path[1].classList.add('parentBoxColor');
-    } else if(e.path[1].id === 'numbers') {
-        return
-    } else {
-        this.classList.add('box-border');
-                
-        let selectedBoxColumn = this.parentNode.firstElementChild;
-        selectedBoxColumn.classList.add('parentBoxColor');
-            
-        if(selectedBoxId == x[selectedBoxId].innerHTML) {
-            x[selectedBoxId].classList.add('parentBoxColor');
+const deselectColumns = (columns) => {
+    const columnCount = columns.length;
+    for (let columnIndex = 0; columnIndex < columnCount; columnIndex++) {
+        const columnCells = columns[columnIndex].children;
+        const cellCount = columnCells.length;
+        for (let cellIndex = 0; cellIndex < cellCount; cellIndex++) {
+            columnCells[cellIndex].classList.remove('selectedHeader');
+            columnCells[cellIndex].classList.remove('selectedCell');
         }
     }
- }; 
-
-
-function deselect(e) {
-    const cols = e.path[2].children;
-    //console.log(e.path[2].firstElementChild.classList);
-    for(let i = 0; i < cols.length; i++) {
-        //console.log();
-        if(cols[i].classList.value === 'box parentBoxColor') {
-            console.log(cols[i]);
-            cols[i].classList.remove('parentBoxColor');
-        };
-
-        let x = cols[i].children;
-
-        for(let j = 0; j < x.length; j++) {
-            if(x[j].classList.value == 'box parentBoxColor' || x[j].classList.value == 'box box-border') {
-                x[j].classList.remove('parentBoxColor');
-                x[j].classList.remove('box-border');
-            }
-        }
-    }; 
 };
-
-
-[...boxes].forEach(box => box.addEventListener('click', deselect));
-[...boxes].forEach(box => box.addEventListener('click', select));
-
-
-//////////////////////////////////////////////////////////////
-
-//create function to connect box with type input
-
-const input = document.getElementById('fxInput');
-const selectedBox = document.getElementsByClassName('box box-border')
-
-function type(e) {
-    //console.log(selectedBox);
-    //console.log(typeof input.value);
-    selectedBox[0].innerHTML = input.value;
+const onCellClicked = (event) => {
+    const selectedCell = event.currentTarget;
+    currentSelectedCell = selectedCell;
+    deselectColumns(selectedCell.parentNode.parentNode.children);
+    selectCell(selectedCell);
 };
```

Comment:
It's a lot easier to follow the logic of the code when each element only has 1 event listener per event type.
Here the `onCellClicked` function is the one taking care of what to do when a cell is clicked.

Another advantage is that it also helps abstract the logic behind a function. If you later want to simplify how the code reacts to a cell being clicked, or even add more functionalities to it, then you just need to change what that function does, but you don't need to change the event listeners added to the element.